fahrschein-http-api/src/main/java/org/zalando/fahrschein/http/api/ContentEncoding.java

```diff
@@ -1,15 +1,42 @@
 package org.zalando.fahrschein.http.api;
 
+import com.github.luben.zstd.ZstdOutputStream;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.zip.GZIPOutputStream;
+
 public enum ContentEncoding {
-    IDENTITY("identity"), GZIP("gzip");
+    IDENTITY("identity"), GZIP("gzip"), ZSTD("zstd");
 
     private final String value;
 
     ContentEncoding(String value) {
         this.value = value;
     }
 
+    private static final Set<String> HTTP_METHODS_WITH_COMPRESSION_SUPPORT = new HashSet<>(Arrays.asList("POST"));
+
+    public boolean isSupported(String httpMethod) {
+        return HTTP_METHODS_WITH_COMPRESSION_SUPPORT.contains(httpMethod);
+    }
+
     public String value() {
         return value;
     }
-}
+
+    public OutputStream wrap(OutputStream out) throws IOException {
+        switch (this) {
+            case GZIP:
+                return new GZIPOutputStream(out);
+            case ZSTD:
+                return new ZstdOutputStream(out);
+            default:
```

Comment:
I think it's better to also have a case for `IDENTITY` and throw an `UnsupportedOperationException` for `default`. It's more future-proof if we someday add new encoding.